pkg/controller/servicemesh/controlplane/readiness_test.go

```diff
@@ -0,0 +1,371 @@
+						Namespace: nonMemberNamespace,
+					},
+				},
+				IngressGateways: map[string]*maistrav2.IngressGatewayConfig{
+					"additional-ingress": {
+						GatewayConfig: maistrav2.GatewayConfig{
+							Namespace: nonMemberNamespace,
+						},
+					},
+				},
+				EgressGateways: map[string]*maistrav2.EgressGatewayConfig{
+					"additional-egress": {
+						GatewayConfig: maistrav2.GatewayConfig{
+							Namespace: nonMemberNamespace,
+						},
+					},
+				},
+			},
+			objects: []runtime.Object{
+				newDeployment("deploy1", nonMemberNamespace, controlPlaneNamespace, "component1", true),
+				newDeployment("deploy2", nonMemberNamespace, controlPlaneNamespace, "component2", true),
+				newDeployment("deploy3", nonMemberNamespace, controlPlaneNamespace, "component3", true),
+				newDeployment("deploy4", nonMemberNamespace, controlPlaneNamespace, "component4", true),
+			},
+			expectedMap: map[string]bool{},
+		},
+		{
+			// components without objects marked as always ready should appear in the map as ready
+			name:                  "always-ready-components",
+			alwaysReadyComponents: "alwaysReady1,alwaysReady2",
+			objects:               []runtime.Object{},
+			expectedMap: map[string]bool{
+				"alwaysReady1": true,
+				"alwaysReady2": true,
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+
+			smcp := newControlPlane()
+			smcp.Spec.Gateways = tc.gateways
+			if tc.alwaysReadyComponents != "" {
+				smcp.Status.Annotations = map[string]string{
+					statusAnnotationAlwaysReadyComponents: tc.alwaysReadyComponents,
+				}
+			}
+
+			cl, tracker := test.CreateClient(tc.objects...)
+			fakeEventRecorder := &record.FakeRecorder{}
+
+			smmr := &maistrav1.ServiceMeshMemberRoll{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "default",
+					Namespace: controlPlaneNamespace,
+				},
+				Spec: maistrav1.ServiceMeshMemberRollSpec{
+					Members: memberNamespaces,
+				},
+				Status: maistrav1.ServiceMeshMemberRollStatus{
+					Members:           memberNamespaces,
+					ConfiguredMembers: memberNamespaces,
+				},
+			}
+			test.PanicOnError(tracker.Add(smmr))
+
+			instanceReconciler := NewControlPlaneInstanceReconciler(
+				common.ControllerResources{
+					Client:            cl,
+					Scheme:            scheme.Scheme,
+					EventRecorder:     fakeEventRecorder,
+					OperatorNamespace: "istio-operator",
+				},
+				smcp,
+				cni.Config{Enabled: true}).(*controlPlaneInstanceReconciler)
+
+			readinessMap, err := instanceReconciler.calculateComponentReadinessMap(ctx)
+			if err != nil {
+				t.Fatalf("Unexpected error in calculateComponentReadinessMap: %v", err)
+			}
+
+			assert.DeepEquals(readinessMap, tc.expectedMap, "Unexpected readiness map", t)
+		})
+	}
+
+}
+
+func newDeployment(name, namespace, owner, component string, ready bool) *appsv1.Deployment {
+	var readyReplicas int32
+	if ready {
+		readyReplicas = 1
+	} else {
+		readyReplicas = 0
+	}
```

Comment:
nit: Zero value for `int32` is by default `0`
```suggestion
	var readyReplicas int32
	if ready {
		readyReplicas = 1
	}
```

Reply:
Same for [this](https://github.com/maistra/istio-operator/pull/795/files#diff-db0700473d726d545874acf32f5a10643272d68e6df319fc276b2e5c84f9e043R325)  and [this](https://github.com/maistra/istio-operator/pull/795/files#diff-db0700473d726d545874acf32f5a10643272d68e6df319fc276b2e5c84f9e043R350).

Reply:
Of course. Thanks!